koku/hcs/csv_file_handler.py

```diff
@@ -33,11 +34,23 @@ def write_csv_to_s3(self, date, data, cols, finalize=False, tracing_id=None):
         filename = f"hcs_{date}.csv"
         month = date.strftime("%m")
         year = date.strftime("%Y")
+        context = {
+            "provider_uuid": self._provider_uuid,
+            "provider_type": self._provider,
+            "schema": self._schema_name,
+            "date": date,
+        }
+        finalize_date = None
+        if finalize:
+            # reports are finalized on the 15th of the month following the report date
+            finalize_date = (date.replace(day=15) + relativedelta.relativedelta(months=1)).strftime("%Y-%m-%d")
```

Comment:
I probably just dont understand exactly what we are trying to pass along here but can you just confirm this is what we expect?
`date`
`datetime.datetime(2023, 8, 10, 12, 44, 48, 534895)` 
`finalize_date = (date.replace(day=15) + relativedelta.relativedelta(months=1)).strftime("%Y-%m-%d")`
`finalize_date`
`'2023-09-15'`

Reply:
Yes this is expected, we are always expecting to send the 15th of the month following the date the report is from. The JIRA ticket has a few more details on the what and a link to the PUDS document on the why, but this is the gist of it:
```
key: x-amz-meta-finalized-date

value-format: YYYY-MM-DD

This date should be set to the 15th of the month following the processed date, even if that file has to be reprocessed for any reason. For example, if the file is for 2023-05-25, the finalized date should be set to 2023-06-15.
```